skyd/cache.go

```diff
@@ -163,6 +164,33 @@ func (psc *PinnedSkylinksCache) threadedRebuild(skydClient Client) {
 		}
 	}
 
+	// If omitBlockedSkylinks is true we'll check all skylinks against the list
+	// of blocked skylinks in skyd and we'll remove the blocked ones from the
+	// cache.
+	if omitBlockedSkylinks {
+		blocklist, err := skydClient.Blocklist()
+		if err != nil {
+			err = errors.AddContext(err, "failed to fetch blocklist")
+			return
+		}
+		var sl skymodules.Skylink
+		for s := range sls {
+			err = sl.LoadString(s)
+			if err != nil {
+				// This is an invalid skylink, remove it.
+				delete(sls, s)
+			}
+			for _, bl := range blocklist.Blocklist {
```

Comment:
Not sure how big the blocklist is at this point but it might be worthwhile to turn this into a map first.

Reply:
Oh, that's true. This might take quite a while otherwise.